bigtable/client/row_range.cc

```diff
@@ -0,0 +1,118 @@
+// Copyright 2017 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "bigtable/client/row_range.h"
+
+#include <cstdlib>
+#include <iostream>
+#include <thread>
+
+#include <absl/memory/memory.h>
+
+namespace bigtable {
+inline namespace BIGTABLE_CLIENT_NS {
+
+const std::string RowRange::EMPTY_KEY = "\377\377\377";;
+// Any range [x, x).
+RowRange RowRange::empty() {
+  return RowRange(RowRange::EMPTY_KEY, RowRange::EMPTY_KEY);
+}
+
+RowRange RowRange::infinite_range() {
+  return RowRange("", "");
+}
+
+// Default one: [begin, end).
+RowRange RowRange::right_open(std::string begin, std::string end) {
+  return RowRange(begin, end);
+}
+
+// (begin, end).
+RowRange RowRange::open_interval(std::string begin, std::string end) {
+  return RowRange(Successor(begin), end);
+}
+
+// [begin, end].
+RowRange RowRange::closed_interval(std::string begin, std::string end) {
+  return RowRange(begin, Successor(end));
+}
+
+// (begin, end].
+RowRange RowRange::left_open(std::string begin, std::string end) {
+  return RowRange(Successor(begin), Successor(end));
+}
+
+RowRange RowRange::prefix_range(std::string prefix) {
+  return RowRange(prefix, Successor(prefix));
+}
+  // An alias for right_open().
+RowRange RowRange::range(std::string begin, std::string end) {
+  return right_open(begin, end);
+}
+
+std::string RowRange::Successor(std::string& key) {
+  if (key == "") {
+    return "";
+  }
+  int position=key.length()-1;
```

Comment:
Spaces around `=` and `-`.

Can you run `clang-format` on this change? It should catch this.